### PR TITLE
Initial Build Ads Performance Script for Rackspace (LinkedIn)

### PR DESCRIPTION
**Pull request by meoflaxz**
Client: Rackspace
Type: Ads Performance Script
Platform: LinkedIn (for now)

Ticket: https://rackspace.wise-portal.com/request-summary.php?id=18

Just an initial build with airtable. Nothing serious

### CODE REVIEW

File: Clients/Rackspace/rackspace_ads_performance.sql [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    one_click_lead_form_opens AS _lead_form_opens,
+    video_starts AS _video_play,
+    video_first_quartile_completions AS _video_views_25percent,
+    video_midpoint_completions AS _video_views_50percent,
+    video_third_quartile_completions AS _video_views_75percent,
+    video_completions AS _video_completions
+  FROM `rackspace_linkedin.ad_analytics_by_creative`
+),
+ads_title AS (
+    SELECT
```
[the remark this answers is not in the document]
Comment (izztsfn) [reply Clients/Rackspace/rackspace_ads_performance.sql:25]: Do the same for all of the remaining CTEs, this and the rest after it bcs they are all not inline. Just reduce one indentation for the content and it should be okay

File: Clients/Rackspace/rackspace_ads_performance.sql [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+total_ads AS (
+    SELECT 
+      *, 
+      COUNT(_ad_id) OVER (PARTITION BY _start_date, _campaign_name ) AS _ads_per_campaign
+    FROM main_data
+),
+daily_budget_per_ad_per_campaign AS (
+    SELECT 
+      *,
+      CASE WHEN _ads_per_campaign > 0 THEN _daily_budget / _ads_per_campaign
```
Comment (izztsfn) [inline Clients/Rackspace/rackspace_ads_performance.sql:104]: Follow this syntax for CASE statement:
```
CASE xxx
  WHEN xxx THEN xxx
  ELSE xx
END AS _xxx
```

File: Clients/Rackspace/rackspace_ads_performance.sql [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+),
+ads_title AS (
+    SELECT
+      SPLIT(SUBSTR(c.id, STRPOS(c.id, 'sponsoredCreative:')+18))[ORDINAL(1)]  AS _adid,
+      campaign_id AS _campaignid,
+      account_id AS _account_id,
+      REGEXP_REPLACE(acc.name, r'[^a-zA-Z]', '') AS _account_name
+    FROM `rackspace_linkedin.creatives` c
+    LEFT JOIN `rackspace_linkedin.accounts` acc 
+        ON acc.id = account_id
```
Comment (izztsfn) [inline Clients/Rackspace/rackspace_ads_performance.sql:32]: reduce on indentation for ON acc.id = account_id

**Review by izztsfn — CHANGES_REQUESTED** [review]

**Comment by meoflaxz** [comment]
Halo @izztsfn,

Just made the required amendment.

Now fixed indentation and uses INSERT instead CREATE OR REPLACE

**Review by hafizasyraf72 — APPROVED** [review]

File: Clients/Rackspace/rackspace_ads_performance.sql [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    video_completions AS _video_completions
+  FROM `rackspace_linkedin.ad_analytics_by_creative`
+),
+ads_title AS (
+  SELECT
+    SPLIT(SUBSTR(c.id, STRPOS(c.id, 'sponsoredCreative:')+18))[ORDINAL(1)]  AS _adid,
+    campaign_id AS _campaignid,
+    account_id AS _account_id,
+    REGEXP_REPLACE(acc.name, r'[^a-zA-Z]', '') AS _account_name
+  FROM `rackspace_linkedin.creatives` c
```
Comment (izztsfn) [inline Clients/Rackspace/rackspace_ads_performance.sql:111]: Remember to use the full table ID here (with the `x-marketing` in front) for readability next time

**Review by izztsfn — APPROVED** [review]
Overall formatting looks good. Some minor comments for next time but i'll still approve this one

**Pull request merged**